llvm/docs/GlobalISel/GenericOpcode.rst

```diff
@@ -521,16 +521,18 @@ The return value of (FMAXNUM 0.0, -0.0) could be either 0.0 or -0.0.
 G_FMINNUM_IEEE
 ^^^^^^^^^^^^^^
 
-Perform floating-point minimum on two values, following the IEEE-754 2008
-definition. This differs from FMINNUM in the handling of signaling NaNs. If one
-input is a signaling NaN, returns a quiet NaN.
+Perform floating-point minimum on two values, following the IEEE-754
+2019 definition. This differs from FMINNUM in the handling of
+signaling NaNs. If one input is a signaling NaN, returns a quiet
+NaN. This treats -0 as ordered less than +0.
```

Comment:
No, because they differ in the behavior for signaling nans. The minnum/maxnum intrinsics were misnamed. They should have been named fmin/fmax, as they follow the libm behavior which ignores signaling nans. The IEEE signaling nan behavior makes no sense, and returns a quiet nan.

Nothing about the nan handling is changing. This PR only adds the stronger signed 0 behavior

Reply:
IEEE-754 2019 defines two functions for minimum: `minimum` and `minimumNumber`. So it worth mentioning that this node represents just `minimumNumber`. The described treatment of SNaN is also slightly different from IEEE-754:
```
minimumNumber(x, y) is x if x < y, y if y < x, and the number if one operand is a number and the
other is a NaN. For this operation, −0 compares less than +0. If x = y and signs are the same it is
either x or y. If both operands are NaNs, a quiet NaN is returned, according to 6.2. If either
operand is a signaling NaN, an invalid operation exception is signaled, but unless both operands
are NaNs, the signaling NaN is otherwise ignored and not converted to a quiet NaN as stated in
6.2 for other operations.
```
Is this difference intentional?

Reply:
Yes. The point of these intermediate nodes is to match hardware behavior for legalization, in which case (for AMDGPU at least) they have the 2008 signaling nan behavior with the stronger 2019 signed zero behavior. 

But for the unconstrained versions, the langref hand waves away signaling nan handling saying they may be treated as quiet nans 

Maybe these nodes need another rename.